src/shogun/multiclass/KNN.cpp

```diff
@@ -575,55 +315,32 @@ void CKNN::store_model_features()
 	SG_UNREF(d_rhs);
 }
 
-int32_t CKNN::choose_class(float64_t* classes, int32_t* train_lab)
+void CKNN::init_solver(KNN_SOLVER knn_solver)
```

Comment:
Hi @vigsterkr , I got **error while running Ctest**[1] and it seems like there are a bunch of errors I haven't meet before[2]. BTW, I run all these test to **virtualbox with ubuntu16.04** due to the valgrind doesn't full support OS X10.12.3[3], is that possible I got **ViennaCL: FATAL ERROR** is because I only give one kernel to virtual machine?

[1] https://pastebin.mozilla.org/8979031
[2] https://pastebin.mozilla.org/8979032
[3] http://valgrind.org/downloads/current.html